### PR TITLE
Move offers to its own module

### PR DESCRIPTION
**Pull request by meaboutsoftware**
This PR includes work related to taking out the logic related to offers and split to 2 own projects:

- Offers API
- Offers DataAccess

### CODE REVIEW

File: Chapter-2-modules-separation/Src/Fitnet.Offers.DataAccess/DataAccessModule.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+namespace EvolutionaryArchitecture.Fitnet.Offers.DataAccess;
+
+using Database;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+public static class DataAccessModule
```
Comment (kamilbaczek) [inline Chapter-2-modules-separation/Src/Fitnet.Offers.DataAccess/DataAccessModule.cs:8]:
```suggestion
internal static class DataAccessModule
```

File: Chapter-2-modules-separation/Src/Fitnet.Offers.DataAccess/Database/DatabaseModule.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
+namespace EvolutionaryArchitecture.Fitnet.Offers.DataAccess.Database;
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+public static class DatabaseModule
```
Comment (kamilbaczek) [inline Chapter-2-modules-separation/Src/Fitnet.Offers.DataAccess/Database/DatabaseModule.cs:8]:
```suggestion
internal static class DatabaseModule
```

**Review by kamilbaczek — CHANGES_REQUESTED** [review]
please make one class internal

File: Chapter-2-modules-separation/Src/Fitnet.Offers.DataAccess/Database/DatabaseModule.cs (hunk 2 above)
Comment (meaboutsoftware) [reply Chapter-2-modules-separation/Src/Fitnet.Offers.DataAccess/Database/DatabaseModule.cs]: Good spot!

Comment (meaboutsoftware) [reply Chapter-2-modules-separation/Src/Fitnet.Offers.DataAccess/Database/DatabaseModule.cs]: Done

**Review by kamilbaczek — APPROVED** [review]

**Pull request merged**